### DISCUSSION TITLE
YD_Lidar x4 compile error

### DISCUSSION THREAD

**Discussion by jarvis2603** [makerspet/support#13] [General]
Arduino: 1.8.19 (Mac OS X), Board: "ESP32 Dev Module, Disabled, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, Core 1, Core 1, None, Disabled"

In file included from /Users/Downloads/firmware-0.7.0/kaiaai-esp32/kaiaai-esp32.ino:26:
/Users/Downloads/meker_pet _pcb/firmware-0.7.0/kaiaai-esp32/lidar.h: In function 'void setupLIDAR()':
lidar.h:223:41: error: expected type-specifier before 'LDS_YDLIDAR_X4_PRO'
                             lidar = new LDS_YDLIDAR_X4_PRO();
                                         ^~~~~~~~~~~~~~~~~~
Multiple libraries were found for "WiFi.h"
 Used: /Users/Library/Arduino15/packages/esp32/hardware/esp32/2.0.11/libraries/WiFi
 Not used: /private/var/folders/g5/j2x_yyzx0kbg75z6sh3h4sjc0000gn/T/AppTranslocation/EA0BF6D4-3597-4355-AE8F-1BF00261AFFB/d/Arduino.app/Contents/Java/libraries/WiFi
exit status 1
expected type-specifier before 'LDS_YDLIDAR_X4_PRO'

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by makerspet** [comment]
Hi, can you please download the latest code (not the latest release - the latest code straight from the repo) and try compiling  again?
I believe this compile error may have been fixed in the latest code
